### DISCUSSION TITLE
Mongo

### DISCUSSION THREAD

**Discussion by vishalduke** [Netflix/conductor#3238] [General]
we want to know whether the Netflix conductor can read data from

Mongo DB
Queue
meaning, task or workflow should read data from the above endpoint while execution of logic

**Comment by v1r3n** [comment]
@vishalduke Yes you can do that.  It is possible to implement a worker (and is a pretty common place to do) that reads data from MongoDB (or any other DB) for that matter.

Take a look at the [Worker](https://github.com/Netflix/conductor/blob/main/client/src/main/java/com/netflix/conductor/client/worker/Worker.java) interface which you will implement that does read data from MongoDB

**Reply by vishalduke** [reply]
@v1r3n 
We need mongo DB as persistence as well

**Reply by maggi1129** [reply]
yes we can do that using the http task, we can use MongoDB Atlas Data APIs

**Reply by vishalduke** [reply]
@maggi1129  we are not able to view the shared link, seems it has restricted access

**Reply by maggi1129** [reply]
@vishalduke 

{
      "name": "get_random_fact",
      "taskReferenceName": "get_random_fact",
      "inputParameters": {
        "http_request": {
          "uri": "<mongoDB Atlas Data API endpoint>",
          "method": "POST",
          "contentType": "application/json",
          "accept": "application/json",
          "connectionTimeOut": 5000,
          "readTimeOut": 5000,
          "headers": {
            "api-key": "<Atlas api key>"
          },
          "body": {
            "collection": "<Your Collection>",
            "database": "<Your DB>",
            "dataSource": "<Your Data Source>"
          }
        }
      },
      "type": "HTTP",
      "decisionCases": {},
      "defaultCase": [],
      "forkTasks": [],
      "startDelay": 0,
      "joinOn": [],
      "optional": false,
      "defaultExclusiveJoinTask": [],
      "asyncComplete": false,
      "loopOver": []
    },
